src/bindings/c/src/ov_c_api.cpp

```diff
@@ -0,0 +1,2052 @@
+    for (auto iter = element_type_map.begin(); iter != element_type_map.end(); iter++) {
+        if (iter->second == type) {
+            return iter->first;
+        }
+    }
+    return ov_element_type_e::UNDEFINED;
+}
+
+#define GET_OV_ELEMENT_TYPE(a)   element_type_map[a]
+#define GET_CAPI_ELEMENT_TYPE(a) find_ov_element_type_e(a)
+
+#define GET_OV_COLOR_FARMAT(a) \
+    (color_format_map.find(a) == color_format_map.end() ? ov::preprocess::ColorFormat::UNDEFINED : color_format_map[a])
+
+#define CATCH_OV_EXCEPTION(StatusCode, ExceptionType) \
+    catch (const InferenceEngine::ExceptionType&) {   \
+        return ov_status_e::StatusCode;               \
+    }
+
+#define CATCH_OV_EXCEPTIONS                                   \
+    CATCH_OV_EXCEPTION(GENERAL_ERROR, GeneralError)           \
+    CATCH_OV_EXCEPTION(NOT_IMPLEMENTED, NotImplemented)       \
+    CATCH_OV_EXCEPTION(NETWORK_NOT_LOADED, NetworkNotLoaded)  \
+    CATCH_OV_EXCEPTION(PARAMETER_MISMATCH, ParameterMismatch) \
+    CATCH_OV_EXCEPTION(NOT_FOUND, NotFound)                   \
+    CATCH_OV_EXCEPTION(OUT_OF_BOUNDS, OutOfBounds)            \
+    CATCH_OV_EXCEPTION(UNEXPECTED, Unexpected)                \
+    CATCH_OV_EXCEPTION(REQUEST_BUSY, RequestBusy)             \
+    CATCH_OV_EXCEPTION(RESULT_NOT_READY, ResultNotReady)      \
+    CATCH_OV_EXCEPTION(NOT_ALLOCATED, NotAllocated)           \
+    CATCH_OV_EXCEPTION(INFER_NOT_STARTED, InferNotStarted)    \
+    CATCH_OV_EXCEPTION(NETWORK_NOT_READ, NetworkNotRead)      \
+    CATCH_OV_EXCEPTION(INFER_CANCELLED, InferCancelled)       \
+    catch (...) {                                             \
+        return ov_status_e::UNEXPECTED;                       \
+    }
+
+char* str_to_char_array(const std::string& str) {
+    std::unique_ptr<char> _char_array(new char[str.length() + 1]);
+    char* char_array = _char_array.release();
+    std::copy_n(str.begin(), str.length() + 1, char_array);
+    return char_array;
+}
+
+const char* ov_get_error_info(ov_status_e status) {
+    if (status > ov_status_e::UNKNOWN_ERROR)
+        return error_infos[ov_status_e::UNKNOWN_ERROR];
+    return error_infos[status];
+}
+
+ov_status_e ov_rank_create(ov_rank_t** rank, int64_t min_dimension, int64_t max_dimension) {
+    if (!rank || min_dimension < -1 || max_dimension < -1) {
+        return ov_status_e::INVALID_PARAM;
+    }
+
+    try {
+        std::unique_ptr<ov_rank_t> _rank(new ov_rank_t);
+        if (min_dimension != max_dimension) {
+            _rank->object = ov::Dimension(min_dimension, max_dimension);
+        } else {
+            if (min_dimension > -1) {
+                _rank->object = ov::Dimension(min_dimension);
+            } else {
+                _rank->object = ov::Dimension();
+            }
+        }
+        *rank = _rank.release();
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
+}
+
+void ov_rank_free(ov_rank_t* rank) {
+    if (rank)
+        delete rank;
```

Comment:
What if delete throws an exception?

Reply:
For C APIs, I don't think it is a must requirement, maybe we cannot pick the bone from the egg, right?